docs/reference/query-dsl/knn-query.asciidoc

```diff
@@ -77,15 +78,17 @@ POST my-image-index/_search
 +
 --
 (Required, string) The name of the vector field to search against. Must be a
-<<index-vectors-knn-search, `dense_vector` field with indexing enabled>>.
+<<index-vectors-knn-search, `dense_vector` field with indexing enabled>>, or a
+<<semantic-text, `semantic_text` field>> with a compatible dense vector inference model.
 --
 
 `query_vector`::
 +
 --
 (Optional, array of floats or string) Query vector. Must have the same number of dimensions
 as the vector field you are searching against. Must be either an array of floats or a hex-encoded byte vector.
-Either this or `query_vector_builder` must be provided.
+Either this or `query_vector_builder` must be provided, unless all queried fields are of type <<semantic-text, semantic_text>>,
+in which case the inference ID associated with the `semantic_text` field will be inferred.
```

Comment:
Should this be a note for `query_vector_builder` rather than `query_vector`? It would be clearer IMO.

Reply:
I think this PR is close to being merged - the only bit is this one.
Looks to me like this note about the inference IDs for the `query_vector_builder` would be better suited under the `query_vector_builder` param?